src/utils/domUtils.ts

```diff
@@ -25,7 +25,7 @@ const updateAttributesFromProps = (
       if (name === 'acceptCharset') {
         name = name.replace(/[A-Z]/g, (match) => `-${match.toLowerCase()}`);
       }
-      element.setAttribute(name, value !== undefined ? value.toString() : '');
+      if (value !== undefined) element.setAttribute(name, value.toString());
```

Comment:
@mmoss 
> and if the value is boolean only set the string if it's a aria- prefixed attribute, otherwise calling removeAttribute

If we check for undefined || null value and remove the attribute first and then in else if block, set the attribute as string for other values which could be - string | boolean | number, this should work as well right? Boolean attributes are set with their names as their value([example](https://github.com/CurriculumAssociates/createjs-accessibility/blob/7da1e20baed2a025f29aa16a77f229206dfa9824/src/RoleObjects/OptionData.js#L25)), so we should be able to set those attributes in the else if block itself.

Is there any particular reason for checking `aria-` prefixed attribute?

Reply:
In general `aria-` prefixed attributes can have the string values "true" and "false," whereas [in HTML the presence or absence](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attributes) of the attribute existing on the DOM element is used instead.

I wasn't sure if we'd diligently enforced the condition in every `*Data.js` implementation like you supplied.